src/shared/components/person/person-listing.tsx

```diff
@@ -24,76 +19,62 @@ export class PersonListing extends Component<PersonListingProps, any> {
   }
 
   render() {
-    const person = this.props.person;
-    const local = person.local;
-    let apubName: string, link: string;
+    const {
+      person: {
+        local,
+        name,
+        actor_id,
+        display_name,
+        published,
+        avatar,
+        banned,
+      },
+      profile,
+    } = this.props;
+    const domain = hostname(actor_id);
+    const apubName = `@${name}@${domain}`;
+    const link = profile ? actor_id : `/u/${name}${!local && `@${domain}`}`;
 
-    if (local) {
-      apubName = `@${person.name}`;
-      link = `/u/${person.name}`;
-    } else {
-      const domain = hostname(person.actor_id);
-      apubName = `@${person.name}@${domain}`;
-      link = !this.props.realLink
-        ? `/u/${person.name}@${domain}`
-        : person.actor_id;
-    }
-
-    let displayName = this.props.useApubName
-      ? apubName
-      : person.display_name ?? apubName;
+    function linkify(node: InfernoNode, isApubName?: boolean) {
+      const className = `text-${display_name || profile ? "muted" : "info"}`;
 
-    if (this.props.showApubName && !local && person.display_name) {
-      displayName = `${displayName} (${apubName})`;
+      if (local && profile) {
+        return isApubName ? <div className={className}>{node}</div> : node;
```

Comment:
This logic wasn't here before, but I figured that there's no point in making it a link when viewing the profile locally because it would just bring the user to the page that they're already on.